api/src/services/listing.service.ts

```diff
@@ -1134,6 +1171,8 @@ export class ListingService implements OnModuleInit {
       dto,
       'listingsBuildingAddress',
     );
+    // Delete all assets tied to listing events before creating new ones
+    await this.updateListingEvents(dto.id);
```

Comment:
This is the way we are handling other sub tables like addresses and paper applications, so I am cool with just deleting and recreating every time. But did you look into only creating if it doesn't exist in the db? Similar to how we do listingsBuildingSelectionCriteriaFile? The logic can get pretty gnarly that way though

Reply:
yeah I think we should address that separately, this is something I think I mentioned but with prisma extensions ( https://www.prisma.io/docs/orm/prisma-client/client-extensions )

we could write a custom query that handles that for us, so we can sort simplify that into helper functions basically 

but I think we should do that in a separate ticketo with a spike to test out extensions a bit